### PR TITLE
Ability to define ID generators per object type via the settings

### PR DESCRIPTION
**Pull request by leplatrem**

### CODE REVIEW

**Comment by leplatrem** [comment]
@glasserc r?

File: docs/configuration/settings.rst [hunk 1]
```diff
@@ -32,36 +32,41 @@ listed below.
...
+|                                                 |              | ``kinto.storage_max_fetch_size`` setting.                                 |
++-------------------------------------------------+--------------+---------------------------------------------------------------------------+
+| kinto.id_generator                              | ``UUID4``    | The Python *dotted* location of the generator class that should be used   |
+|                                                 |              | to generate identifiers on a POST on a records endpoint.                  |
++-------------------------------------------------+--------------+---------------------------------------------------------------------------+
+| kinto.<object-type>_id_generator                | ``UUID4``    | The Python *dotted* location of the generator class that should be used   |
+|                                                 |              | to generate identifiers on a POST endpoint.                               |
+|                                                 |              | Object type is one of ``bucket``, ``collection``, ``group``, ``records``. |
```
Comment (leplatrem) [inline docs/configuration/settings.rst]: typo: `record`

File: kinto/core/initialization.py [hunk 2]
```diff
@@ -176,8 +177,15 @@ def eos_tween(request):
 def setup_storage(config):
     settings = config.get_settings()
 
-    id_generator = config.maybe_dotted(settings['id_generator'])
-    config.registry.id_generator = id_generator()
+    # Id generators by resource name.
+    config.registry.id_generators = {}
+    for key, value in settings.items():
+        m = re.match(r'^([^_]*)_?id_generator', key)
```
Comment (glasserc) [inline kinto/core/initialization.py]: Why not just `key.endswith('id_generator')`?

Comment (glasserc) [inline kinto/core/initialization.py]: Oh, never mind, because you want to retrieve the prefix.

**Pull request merged**